lib/src/inline_syntaxes/footnote_ref_syntax.dart

```diff
@@ -0,0 +1,65 @@
+import '../ast.dart' show Element, Node, Text;
+import '../charcode.dart';
+import 'link_syntax.dart' show LinkContext;
+
+class FootnoteRefSyntax {
```

Comment:
Could you add comments here about where this syntax is specified? Could just be live experiments from GitHub, if it is not spelled out in GitHub-flavored Syntax.

Reply:
It is a handling branch of link syntax, but I move the code into a separated class.

Reply:
Please include a more full comment here. Something like "A syntax for defining a footnote reference, as specified at..." and include a click-able link.

Reply:
resolved in a8a81fc

Reply:
Please address the above comment:

> Please include a more full comment here. Something like "A syntax for defining a footnote reference, as specified at..." and include a click-able link.

A click-able link would probably have a URL starting with 'https://'

Reply:
footnote reference here is a handling branch of `LinkSyntax`, not a kind of syntax. I just moved these code into a separated class, should I rename the class to avoid misunderstanding?

Reply:
Hmm no I don't think it needs to be renamed. Maybe it can just reference `[FootnoteDefSyntax]` so that there is a documentation link between the two. Then it's fine. Thanks!